src/main/webapp/app/shared/markdown-editor/commands/orderedListCommand.ts

```diff
@@ -8,54 +8,62 @@ export class OrderedListCommand extends Command {
 
     /**
      * @function execute
-     * @desc Use the markdown language for creating/removing an ordered list
+     * @desc Use the Markdown language for creating/removing an ordered list
      */
     execute(): void {
         const selectedText = this.getSelectedText();
         this.splitText(selectedText);
     }
 
     /**
+     * Splits the text and performs the necessary manipulations.
      * @function splitText
-     * @desc 1. Split the text at the line break into an array
-     *       2. Assign each line the position it has in the array
-     *       3. Call for each textLine the replaceText method
      * @param selectedText the selected text by the cursor
      */
     splitText(selectedText: string): void {
         const parseArray = selectedText.split('\n');
-        let addAmount = parseArray.length - 1;
-        for (const element of parseArray) {
-            this.replaceText(element, parseArray.length - addAmount);
-            addAmount--;
+        let manipulatedText = '';
+        let position = 1;
+        for (const line of parseArray) {
+            // Empty line -> we start again
+            if (line === '') {
+                // Special case: Single empty line
+                if (parseArray.length === 1) {
+                    manipulatedText = '1. ';
+                    continue;
+                }
+                position = 1;
+            } else {
+                // Manipulate the line, e.g. remove the number or add the number.
+                manipulatedText += this.manipulateLine(line, position);
+                position++;
+            }
+            manipulatedText += '\n';
```

Comment:
This adds an additional newline after the selection when using the command. It’s not a significant issue, but just slightly unexpected behaviour.

Reply:
Yes that was the old behavior - removed now :+1